### PR TITLE
[STUDIO-1820] Security vulnerability: Allow to set Team Domain

### PR DESCRIPTION
**Pull request by avokhmin**
## Security vulnerability: Allow to set Team Domain

[Jira Story](https://shuttlerock.atlassian.net/browse/STUDIO-1820)

The [Cloudflare docs](https://developers.cloudflare.com/cloudflare-one/identity/users/validating-json) suggest constructing the domain form a configuration parameter rather than relying on the inbound Host header

## How to test the PR

- deploy app & her that all works.

## Deployment Notes

No

### CODE REVIEW

**Comment by avokhmin** [comment]
CC @JackDanger

File: lib/rack/cloudflare_jwt/auth.rb [hunk 1]
```diff
@@ -35,21 +33,25 @@ class DecodeTokenError < StandardError; end
...
     #   )
     #
-    # @param policies [Hash<String, String>] the policies with paths and AUDs.
-    def initialize(app, policies = {})
-      @app      = app
-      @policies = policies
+    # @param [Hash<String, String>] options the options to create a middleware.
+    # @option options [Symbol] :team_domain The Team Domain (e.g. 'test.cloudflareaccess.com')
+    # @option options [String] '<path>' The AUD for current '<path>'.
+    def initialize(app, options = {})
```
Comment (avokhmin) [inline lib/rack/cloudflare_jwt/auth.rb]: I thought about:

```suggestion
    def initialize(app, team_domain:, **policies)
```

but ruby 2.6 can't handle it :(

Comment (avokhmin) [reply lib/rack/cloudflare_jwt/auth.rb]: or maybe let's use 

```suggestion
    def initialize(app, team_domain, policies = {})
```

?

Comment (dperrett) [reply lib/rack/cloudflare_jwt/auth.rb]: I like `initialize(app, team_domain, policies = {})`, since the domain is required 👍

Comment (avokhmin) [reply lib/rack/cloudflare_jwt/auth.rb]: I will update.

**Comment by avokhmin** [comment]
@dperrett PR updated.

**Review by dperrett — APPROVED** [review]
👍

**Pull request merged**